pkg/cloud/gcp/gcp.go

```diff
@@ -54,9 +54,24 @@ type Config struct {
 
 // RunOn runs the given function on GCP, supplying it with a cloud instance connected to GCP and a reporter that writes to CLI.
 // The functions makes sure that infraID and region are specified, and extracts the credentials from a secret in order to connect to GCP.
-func RunOn(restConfigProducer *restconfig.Producer, config *Config, creds *google.Credentials, status reporter.Interface,
+func RunOn(restConfigProducer *restconfig.Producer, config *Config, status reporter.Interface,
 	function func(api.Cloud, api.GatewayDeployer, reporter.Interface) error,
 ) error {
+	var err error
+	if config.OcpMetadataFile != "" {
+		config.InfraID, config.Region, config.ProjectID, err = ReadFromFile(config.OcpMetadataFile)
```

Comment:
We can make this function non-public.

```suggestion
		config.InfraID, config.Region, config.ProjectID, err = readFromFile(config.OcpMetadataFile)
```

Reply:
Just imitated what `AWS` is currently doing.

Reply:
yeah that's fine - we can consolidate the `ReadFromFile` functions later.